### PR TITLE
MJ merge into master

### PR DESCRIPTION
**Pull request by jlipps**
making this so I can add comments @moizjv @Jonahss

### CODE REVIEW

File: lib/android-helpers.js [hunk 1]
```diff
@@ -118,7 +119,7 @@ helpers.getLaunchInfoFromManifest = async function (adb, opts) {
   return {appPackage, appWaitPackage, appActivity, appWaitActivity};
 };
 
-helpers.getRemoteApkPath = async function (localApkMd5) {
+helpers.getRemoteApkPath = function (localApkMd5) {
```
Comment (jlipps) [inline lib/android-helpers.js]: this needs to be `async`

File: lib/driver.js [hunk 2]
```diff
@@ -1,9 +1,102 @@
...
+    let launchInfo = await helpers.getLaunchInfo(this.adb, this.opts);
+    Object.assign(this.opts, launchInfo);
+
+    // install app
+    await helpers.installApkRemotely(this.adb, this.opts.app, this.opts.appPackage, this.opts.fastReset);
+
+    return launchInfo;
+  }
+
+  async initAndroidBootstrap () {
```
Comment (jlipps) [inline lib/driver.js]: do we use this more than once? if not, do we need a 2-line function?

File: lib/driver.js [hunk 3]
```diff
@@ -1,9 +1,102 @@
...
+    this.bootstrap = new Bootstrap(DEVICE_PORT);
+
+    await this.bootstrap.start(this.opts.appPackage, false);
+  }
 
+  async startAUT () {
```
Comment (jlipps) [inline lib/driver.js]: do we need a 1-line function? do we use this more than once?

Comment (Jonahss) [inline lib/driver.js]: moiz made this function larger i think. lets wait till the end to refactor the flow.

**Pull request merged**